tools/scripts/update-bumpenvs-yaml.py

```diff
@@ -0,0 +1,213 @@
+    # We only care about artifact lines with exactly six fields.
+    ArtifactLine = collections.namedtuple(
+        "ArtifactLine", "time builder linetype index msgtype val"
+    )
+
+    # We only care about artifact lines with exactly 6 fields.
+    artifact_lines = [
+        ArtifactLine(*f) for f in fields if len(f) == 6 and f[2] == "artifact"
+    ]
+
+    # Get the ami images, which should match lines like this one (line break for readability):
+    #   1598642161,amazon-ebs,artifact,0,id,us-east-1:
+    #       ami-04894a7352df9fdf9%!(PACKER_COMMA)us-west-2:ami-017627938fe327e4f
+    ami_lines = [
+        a
+        for a in artifact_lines
+        if a.builder == "amazon-ebs" and a.msgtype == "id" and a.val
+    ]
+    assert (
+        len(ami_lines) == 1
+    ), f"expected one matching ami ids line but got: {ami_lines}"
+
+    ami_fields = ami_lines[0].val.split("%!(PACKER_COMMA)")
+    for ami_field in ami_fields:
+        region, ami = ami_field.split(":")
+        name = region.replace("-", "_") + "_agent_ami"
+        out[name] = ami
+
+    # Get the GCP builder name by matching the build-id from a line like this one:
+    #   1598642161,det-environments-06318c7,artifact,0,builder-id,packer.googlecompute
+    gcp_builders = [
+        a.builder
+        for a in artifact_lines
+        if a.msgtype == "builder-id" and a.val == "packer.googlecompute"
+    ]
+    assert len(gcp_builders) == 1, f"expected one gcp builder but got: {gcp_builders}"
+    gcp_builder = gcp_builders[0]
+
+    # Get the GCP artifact ID by matching a line like this one:
+    #    1598642161,det-environments-06318c7,artifact,0,id,det-environments-06318c7
+    gcp_ids = [
+        a.val
+        for a in artifact_lines
+        if a.builder == gcp_builder and a.msgtype == "id" and a.val
+    ]
+    assert len(gcp_ids) == 1, f"expected one matching gcp id line but got: {gcp_ids}"
+    out["gcp_env"] = gcp_ids[0]
+
+    return out
+
+
+def update_tag_for_image_type(subconf: Dict[str, str], new_tag: str) -> bool:
+    if new_tag == subconf["new"]:
+        return False
+
+    subconf["old"] = subconf["new"]
+    subconf["new"] = new_tag
+    return True
+
+
+if __name__ == "__main__":
+    if len(sys.argv) < 3:
+        print(__doc__, file=sys.stderr)
+        sys.exit(1)
+
+    path = sys.argv[1]
+    commit = sys.argv[2]
+
+    with open(path) as f:
+        conf = yaml.safe_load(f)
+
+    builds = get_all_builds(commit)
+    artifacts = get_all_artifacts(builds)
+
+    new_tags = {
+        **yaml.safe_load(artifacts["publish-tf1-cpu"]),
+        **yaml.safe_load(artifacts["publish-tf1-gpu"]),
+        **yaml.safe_load(artifacts["publish-tf2-cpu"]),
+        **yaml.safe_load(artifacts["publish-tf2-gpu"]),
+        **parse_packer_log(artifacts["packer-log"]),
+    }
+
+    saw_change = False
+    for image_type, new_tag in new_tags.items():
+        saw_change |= update_tag_for_image_type(conf[image_type], new_tag)
```

Comment:
D'oh :man_facepalming: . It feels a bit unusual to use `|` for `bool`s in Python, but I guess it works.